src/html.rs

```diff
@@ -221,3 +222,45 @@ pub fn select(el: &NodeRef, selectors: &str) -> impl Iterator<Item = NodeRef> {
         .unwrap()
         .map(|el_data| el_data.as_node().clone())
 }
+
+fn scoping_nodes<F>(el: &NodeRef, filter_fn: F) -> impl Iterator<Item = NodeRef>
+where
+    F: Copy + Fn(&NodeRef) -> bool,
+{
```

Comment:
huh, this looks a bit expensive... Anyhow we can profile as needed. For now this is fine